### ISSUE TITLE
Surface more information about widgets

### ISSUE THREAD

**Issue by bernars-usa** [usds/us-forms-system#138]
From https://github.com/usds/us-forms-system/pull/112#discussion_r198970679:

> (In ), I copied these from `/src/js/widgets`, but need some help filling out the info in the second column. I'm also not positive that these are all intended to be listed. I'm tempted to format these similarly to the definitions reference information below, mostly because they could possibly need descriptions. Reading the names is fairly self-explanatory, but I'm guessing there's some unique information about how these are implemented that might be useful to describe.

@annekainicUSDS very helpfully responded:

> Yes, I agree, there's more info we can capture here that would be useful. I think the second column should rather describe how to use each one. For the ones below where you've already listed something in the second column, those widgets are used by default when the schema looks a certain way (e.g., `CheckboxWidget` is used when the schema has `type: boolean`). For the others, they can either be set directly by doing `ui:widget` in the schema, or used through an existing definition (e.g, for `PhoneWidget`, it can be used by used the definition for phone, under `definitions/phone.js`.

Just capturing this in a new issue to address in a follow-up PR.

**Comment by dmethvin-gov** [comment]
I'm wondering where this stands as far as priorities go since we talked this morning about not exposing  definitions in us-forms-system (at least for now). Should we wait on widgets as well? The WIP in #163 depends on this answer.

In the meantime I had started building out a `formConfig` that didn't have any external dependencies, it's cloned from the starter app:

https://github.com/dmethvin-gov/awesome-form/blob/4ef7fcecc0410374df7c13520466d8fba92c3f0e/js/config/form.js

**Comment by annekainicUSDS** [comment]
Even if we pull out definitions, I think we would still want to include widgets in the library for the hackathon. Many of the widgets are used as the default fields to render for generic fields, like `TextWidget` and `SelectWidget`. The definitions are rather common pieces of the `formConfig` that you might extract out because you would reuse it in multiple places, like `fullName` and `address`, but they really depend still on the specific way you ask those questions in your application. For example, for Vets.gov we decided that for every place we ask for someone's name we include 4 fields: first, middle, last, and suffix. It's entirely possible a user of the library wouldn't want to include a field for suffix, and therefore the definition would be unusable by them.

I've been thinking about the most useful way to include helpers for common fields, and I feel like that could use some more thinking around. As they are now, I'm just not sure how useful definitions will be to people :/. But most of the widgets definitely are!

**Comment by dmethvin-gov** [comment]
Sounds good, I'll work on widgets then.

**Comment by annekainicUSDS** [comment]
Marking this as complete! With this PR https://github.com/usds/us-forms-system/pull/166 and this PR https://github.com/usds/us-forms-system/pull/174
